### PR TITLE
Modify docker inspect client to check statusCode instead of strings c…

### PR DESCRIPTION
**Pull request by haoshuwei**
Modify docker client "docker network inspect" to check statusCode instead of strings contains check

Signed-off-by: Shuwei Hao haosw@cn.ibm.com

### CODE REVIEW

File: api/client/inspect.go [hunk 1]
```diff
@@ -60,38 +62,37 @@ func (cli *DockerCli) CmdInspect(args ...string) error {
...
 				}
 				if *inspectType == "container" {
-					if strings.Contains(err.Error(), "No such") {
-						fmt.Fprintf(cli.err, "Error: No such container: %s\n", name)
-					} else {
+					if statusCode != http.StatusNotFound {
 						fmt.Fprintf(cli.err, "%s", err)
 					}
 					status = 1
+					fmt.Fprintf(cli.err, "Error: No such container: %s\n", name)
```
Comment (vdemeester) [inline api/client/inspect.go]: Shouldn't this one be in a `else` ? (otherwise we're gonna print it for any error, which it does not now)

**Comment by haoshuwei** [comment]
@vdemeester ,yes ,you are right,it should be in an else

File: api/client/inspect.go [hunk 2]
```diff
@@ -60,16 +62,16 @@ func (cli *DockerCli) CmdInspect(args ...string) error {
...
 					return err
 				}
 				if *inspectType == "container" {
-					if strings.Contains(err.Error(), "No such") {
-						fmt.Fprintf(cli.err, "Error: No such container: %s\n", name)
-					} else {
+					if statusCode != http.StatusNotFound {
 						fmt.Fprintf(cli.err, "%s", err)
+					} else {
+						fmt.Fprintf(cli.err, "Error: No such container: %s\n", name)
```
Comment (thaJeztah) [inline api/client/inspect.go]: Just a nit; I think the if/else should be swapped, I.e. check for `statusCode == http.StatusNotFound`. It's more consistent with other if/elses in this file.

**Comment by haoshuwei** [comment]
@thaJeztah ,i have modified  code to swap the print content under if and else

**Comment by thaJeztah** [comment]
Thanks @haoshuwei, LGTM

**Comment by vdemeester** [comment]
LGTM 🐭

**Comment by tiborvass** [comment]
LGTM

**Comment by duglin** [comment]
much better check - thanks!
LGTM

**Comment by duglin** [comment]
windows CI is having issues,ignoring.

**Pull request merged**